tests/udf_communication/socket_factory/test_zmq_poller.py

```diff
@@ -0,0 +1,31 @@
+import zmq
+
+from exasol_advanced_analytics_framework.udf_communication.socket_factory.abstract_socket_factory import SocketType, \
+    PollerFlag
+from exasol_advanced_analytics_framework.udf_communication.socket_factory.zmq_socket_factory import ZMQSocketFactory, \
+    ZMQPoller
+
+
+def test_create_poller():
+    with  zmq.Context() as context:
+        factory = ZMQSocketFactory(context)
+        with factory.create_socket(SocketType.PAIR) as socket1, \
+                factory.create_socket(SocketType.PAIR) as socket2, \
+                factory.create_socket(SocketType.PAIR) as socket3, \
+                factory.create_socket(SocketType.PAIR) as socket4:
+            socket1.bind("inproc://test1")
+            socket2.connect("inproc://test1")
+            socket3.bind("inproc://test2")
+            socket4.connect("inproc://test2")
+            socket1.send(b"123")
+            socket3.send(b"456")
+            poller = factory.create_poller()
+            poller.register(socket1, flags=PollerFlag.POLLIN)
+            poller.register(socket2, flags=PollerFlag.POLLIN)
+            poller.register(socket3, flags=PollerFlag.POLLIN)
+            poller.register(socket4, flags=PollerFlag.POLLIN)
+            result = poller.poll()
+            assert result == {
+                socket2: {PollerFlag.POLLIN},
+                socket4: {PollerFlag.POLLIN}
+            }
```

Comment:
With an [ExitStack](https://docs.python.org/3/library/contextlib.html?highlight=context%20stack#contextlib.ExitStack) you may be able to simplify or remove some repition.

sudo code:

```python
def test_...(addresses):
    ...
    factory = ZMQSocketFactory(context)
    with ExitStack() as stack:
        sockets = [stack.enter_context(factory.create_socket(SocketType.PAIR)) for _ in addresses]
        ...

```

Reply:
These sockets are not a file descriptor, or at least not only, you might produce inconsistent object with your example

Reply:
ah, ok yeah this could work, understood it incorrectly first.

Reply:
But, to be honest, it introduces for 4 sockets a mechanic you need to understand. If we had more sockets, maybe 10, I would agree.